src/main/archetype/ui.frontend.angular/README.md

```diff
@@ -40,3 +40,134 @@ Furthermore, an AEM ClientLib is generated from the app using the [`aem-clientli
+* `npm run deploy-ssr-ioruntime` - Deploys the serverless code to your IO runtime namespace. Requires the environment variables AIO_RUNTIME_NAMESPACE and AIO_RUNTIME_AUTH to be set.
+
+${hash}${hash}${hash}${hash} Notes
+
+To summarize, the way this works:
+
+${hash}${hash}${hash}${hash}${hash} 1: the servlet (in the body.html) will call the serverless endpoint on a page request, sending over the page data using sling model exporters as JSON
+${hash}${hash}${hash}${hash}${hash} 2: the serverless code consumes the JSON and returns pre-rendered HTML over
+${hash}${hash}${hash}${hash}${hash} 3: the servlet puts this inside the HTML
+${hash}${hash}${hash}${hash}${hash} 4: the angular app detects the pre-rendered HTML and hydrates the app instead of rendering it:
+
+${hash}${hash}${hash}${hash} Details:
+
+The server-side rendering gets kicked off by the following SLY statement in your body.html of the page:
+```html
+<!-- this starts the SSR rendering chain !-->
+<div id="spa-root" data-sly-resource="${symbol_dollar}{resource @ resourceType='cq/remote/content/renderer/request/handler'}"></div>
+
+```
+
+This will trigger the out of the box remote renderer.
+The renderer is configured by the following configuration:
+
+com.adobe.cq.remote.content.renderer.impl.factory.ConfigurationFactoryImpl~mysite.cfg
+
+```json
+{
+    "getContentPathPattern": "/content/mysite/(.*)|/conf/mysite/(.*)/settings/wcm/templates/(.*)",
+    "getRemoteHTMLRendererUrl" : "http://localhost:3233/api/v1/web/guest/angular-server-1.1.2/ssr",
+    "getRequestTimeout": 10000,
+    "getAdditionalRequestHeaders": [
+    
+    ]
+}
+```
+
+In here you put the correct endpoint. By default for demonstration purposes the endpoint is configured to your localhost.
+So to get a demo quickly just run the `npm run run-local` script after a full installation.
+Refresh the page and you should see the pre-rendered HTML in the page source.
+
+
+For production purposes the URL should point to an actual IO runtime instance.
+Under "getAdditionalRequestHeaders" of this configuration you can put an authentication header that is needed for IO Runtime, or any other static headers that are desired.
+
+Production:
+
+```json
+{
+    "getContentPathPattern": "/content/mysite/(.*)|/conf/mysite/(.*)/settings/wcm/templates/(.*)",
+    "getRemoteHTMLRendererUrl" : "https://adobeioruntime.net/api/v1/web/<your-namespace-here>/angular-server-1.1.2/ssr",
+    "getRequestTimeout": 10000,
+    "getAdditionalRequestHeaders": [
+        "Auth: Basic <your key here>"
+    ]
+}
+```
+
+Finally, the initialization script of the app is different:
+
+main.ts:
+```js
+const initialStateScriptTag = document.getElementById('__AEM_STATE__');
+if(!!initialStateScriptTag) {
+  try {
+    const initialState = JSON.parse(initialStateScriptTag.innerHTML);
+    // @ts-ignore
```

Comment:
I guess it's fine to remove @ts-ignores from examples ?

Reply:
nope, it won't compile :(